[doc-sprint] Add some narrative text to landing page

alasdairwilson opened the pull request:
A very quick attempt  (slowed by everything having being renamed when rebasing) at solving the same problem as #6615 but in a different way. This has the same information as the current landing page but presented in a more descriptive manner.  

I envision the flashy, attention grabbing content suggested in that issue as going in a new "what is sunpy" page linked right at the start here such that the first steps section goes "what is sunpy -> installation guide -> tutorial/walkthrough"

I think having laid it out like this it highlights the need for the tutorial to be improved and reorganised.

There is some placeholder code-comments envisioning where links to future envisioned sections would go. The whole thing Very much up for feedback.

dstansby commented on docs/index.rst:
```diff
@@ -2,42 +2,64 @@
+``sunpy`` is a community-developed, free and open-source solar
 data analysis environment for Python.
 
+New Users
```
I think this would be better at one title level lower, to make sure "sunpy core Documentation" stays the biggest at the top.

dstansby commented on docs/index.rst:
```diff
@@ -2,42 +2,64 @@
+..
+        * :Introduction to sunpy core [new page described in #6615]
+
+* :doc:`installation guide <intro/installation>`
```
```suggestion
* :doc:`Installation guide <intro/installation>`
```

Cadair replied:
We should avoid using the `:doc:` role where possible and use `:ref:` with the reference names of the titles we are referencing. This is so that if we change the path the links still work.

dstansby commented on docs/index.rst:
```diff
@@ -2,42 +2,64 @@
+
+* :doc:`installation guide <intro/installation>`
+
+Our **tutorial** provides a walkthrough on how to use the key features of sunpy core, we recommend new users start here.
```
But you've just said first time users go to the installation guide 😆 How about:
```suggestion
Our **tutorial** provides a walkthrough on how to use the key features of sunpy core.
Once you've installed sunpy, start here.
```

dstansby commented on docs/index.rst:
```diff
@@ -2,42 +2,64 @@
+
+Our **tutorial** provides a walkthrough on how to use the key features of sunpy core, we recommend new users start here.
+
+* :doc:`Getting Started <intro/index>`
```
I'm not a fan of this manual ordering and then a hidden toc tree. How about we move "Installation guide" to a new page that shows up in the side bar, then include that as a toctree above, and turn this getting started into a toctree?

Cadair replied:
See my comment on the main thread. I actually think there's potential value in the separation.

In this case I think not having installation guide in the sidebar is an improvement (I got rid of it in #6613)

dstansby commented on docs/index.rst:
```diff
@@ -2,42 +2,64 @@
+**********
+
+* :doc:`Example gallery <generated/gallery/index>` contains short-form guides on accomplishing common tasks using sunpy.
+* :doc:`API reference<reference/index>` contains a technical description of the functionality of sunpy core's modules. It describes how they work and how to use them but assume you have a good understanding of sunpy and the scientific python ecosystem in general.
```
```suggestion
* :doc:`API reference<reference/index>` contains a technical description of the inputs, outputs, and behaviour of each component of sunpy core.
```
I don't think API ref requires a good understanding of sunpy? With a bit of knoweldge (but not good understanding) I'd expect to be able to understand API reference entries in isolation.

dstansby commented on docs/index.rst:
```diff
@@ -2,42 +2,64 @@
+..
+    * [we dont have a link to these but we do have some explanation pages] Topic guides (name?) provide
+
+Further Info
+************
+
+* Find out :doc:`what's new <whatsnew/index>` with sunpy.
+* View a list of :doc:`known issues <reference/known_issues>` or report new issues on our `GitHub issue tracker`_.
+* Join our `chat room`_ or post to our `discourse`_ for help.
+* Read the :doc:`Developers guide <dev_guide/index>` to start contributing to the sunpy project.
```
This is excellent 💯

dstansby reviewed (commented):
Nice! My general comment is I think we should try and mirror the structure of the landing page with our toctree, and then include toctrees in the actual landing page instead of having a hidden toctree.

Cadair commented:
> My general comment is I think we should try and mirror the structure of the landing page with our toctree, and then include toctrees in the actual landing page instead of having a hidden toctree.

I am unconvinced by this as a pattern (although I am very open to be proven wrong). The toctree(s) primary function is to determine the layout of the left nav bar. I don't see any reason why the presentation of the content on the landing page has to exactly mirror the links we want to show and the order we want to show them in the sidebar? We might want them to be the same or we might not (now or in the future) so why not keep them separate and keep the option?

Cadair commented on docs/index.rst:
```diff
@@ -2,42 +2,64 @@
+If you are a first time user or new to Python,
+
+..
+        * :Introduction to sunpy core [new page described in #6615]
```
There is actually a sphinx extension which supports todo items and tracking them, might be worth a look: https://www.sphinx-doc.org/en/master/usage/extensions/todo.html ?

alasdairwilson commented:
> I am unconvinced by this as a pattern (although I am very open to be proven wrong). The toctree(s) primary function is to determine the layout of the left nav bar. I don't see any reason why the presentation of the content on the landing page has to exactly mirror the links we want to show and the order we want to show them in the sidebar? We might want them to be the same or we might not (now or in the future) so why not keep them separate and keep the option?

So I hadn't realised you had done the same thing in #6613 but we definitely reached the same conclusion. We currently have a table of links (sidebar) and we present the same data in (two) tables on the page, the idea here was to retain the useful sidebar nav but to present the same information in a different way on the page. I agree this retains more flexibility.

alasdairwilson commented:
So I tried to address the concerns from the code review:

Did all the small style changes form @dstansby and replaced docs with refs for all but the example gallery.

I addressed the concerns over manually populating the tutorial subsections by adding them as a toctree, this means changes to the tutorial will automatically update the landing page.

There  are also 2 additional commits that can be taken or left (or cherry picked for another PR):

One removes the installation guide from the tutorial and instead adds it to the blurb at the beginning of the guide, removing it from the tutorial toctree on the landing page (which also removes it from the sidebar). I am torn on this, think neither way is ideal but we retain the ability to manually populate the sidebar by readding it as a hidden toctree or just removing that last commit which will mean the installation guide appears in the sidebar of /guide and the tutorial toctree on the landing page.

Added Sphinx-ext-todo which, to me looks useful, add a `.. todo::` tag, the same as a `.. note::` tag and it will not render unless you set `todo_include_todos` i.e.  `sphinx-build -D todo_include_todos=1 -b html . _build/html`

This generates this really cool page that I have put at html/todo.html :

![image](https://user-images.githubusercontent.com/60351846/203568762-0e4b1603-81bc-4ed0-ac02-9e33f55ecb2e.png)

I think this could be useful, especially during a time when the docs are in flux as reminders of parts that need completing or parts that will need revisited (as in the case) once other tasks are completed.

Cadair commented:
> One removes the installation guide from the tutorial and instead adds it to the blurb at the beginning of the guide, removing it from the tutorial toctree 

Having the install guide not appear in the menus, either at top level or in the tutorial, seems weird. What's the thinking for not having it in the tutorial toctree?

Cadair commented on docs/index.rst:
```diff
@@ -2,42 +2,74 @@
+If you are a first time user or new to Python, our **tutorial** provides a walkthrough on how to use the key features of sunpy core.
+Once you've installed sunpy, start here.
+
+* :ref:`guide`
```
I thought you had put a toctree here?

alasdairwilson replied:
see my comment in main thread, because of #6616 it now breaks the toctree here

alasdairwilson commented:
Think that's it, sadly the changes in #6616 meant the solution to auto generating a tutorial index on this page no longer works. I suspect there is probably a good solution along the lines of having the layout of the tutorial stored in a separate file and included in both places, preventing them getting out of sync later, but out there but out of scope I think here.

nabobalis commented on docs/index.rst:
```diff
@@ -2,42 +2,74 @@
 
-The sunpy core package is a community-developed, free and open-source solar
-data analysis environment for Python.
+``sunpy`` is a community-developed, free and open-source solar data analysis environment for Python.
```
Do we want to enforce all sunpy library references are marked up like this?

dstansby replied:
I think it looks good here - if we want to disucss enforcing (or not) as a rule, worth opening a new issue?

alasdairwilson replied:
I had a look over the docs and its a bit all over the place how sunpy is stylised: ``sunpy``, SunPy, sunpy core (package), the sunpy project etc.

If there are rules to follow or rules to be made then we should do that. I don't care how it is stylised here really.

nabobalis replied:
We need to enforce this down the road but for now, we don't need to block.

nabobalis commented on docs/index.rst:
```diff
@@ -2,42 +2,74 @@
+* Find out :ref:`what's new <whatsnew>` with sunpy.
+* View a list of :ref:`known issues <known_issues>` or report new issues on our `GitHub issue tracker`_.
+* Join our `chat room`_ or post to our `discourse`_ for help.
+* Read the :ref:`Developers guide <dev_guide>` to start contributing to the sunpy project.
```
```suggestion
* Read the :ref:`Developers guide <dev_guide>` to start contributing to the SunPy project.
```

nabobalis commented on docs/index.rst:
```diff
@@ -2,42 +2,74 @@
+For help installing ``sunpy`` refer to our :ref:`installation guide <installing>`.
+
+
+.. todo::
```
Are we ok with adding todos in files? 

Would an issue not be a better idea?

dstansby replied:
I'm 👍 to opening an issue too, so we don't forget.

dstansby commented:
:shipit:
